File: turi-ui/src/types/attraction.ts
export interface Attraction {
    readonly attractionId?: number
    touristicPlaceId: number
    attractionType: AttractionType
    name: string
    description: string
    price: number
    priceType: PriceType
    prepayment: boolean
    cancelReservationDays?: number
    maxPeopleNumber?: number
    maxItems?: number
    dateFrom: Date
    dateTo: Date
    hourFrom: string
    hourTo: string
    daysReservationBefore: number
}

export enum AttractionType {
    UNASSIGNED = 'UNASSIGNED',
    RELAX = 'RELAX',
    SPORT = 'SPORT',
    RECREATION = 'RECREATION',
    ENTERTAINMENT = 'ENTERTAINMENT',
    FOOD = 'FOOD',
    EVENT = 'EVENT',
    CHILDREN = 'CHILDREN',
    OTHER = 'OTHER'
}

export enum PriceType {
    UNASSIGNED = 0,
    HOUR = 1,
    PERSON = 2,
    ITEM = 3
}

File: turi/src/main/java/com/turi/attraction/domain/model/Attraction.java
package com.turi.attraction.domain.model;

import com.turi.attraction.infrastructure.adapter.repository.AttractionEntity;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;

@Getter
@Setter
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
@Builder(setterPrefix = "with")
public final class Attraction
{
    private Long attractionId;
    private Long touristicPlaceId;
    private AttractionType attractionType;
    private String name;
    private String description;
    private BigDecimal price;
    private PriceType priceType;
    private Boolean prepayment;
    private Integer cancelReservationDays;
    private Integer maxPeopleNumber;
    private Integer maxItems;
    private LocalDate dateFrom;
    private LocalDate dateTo;
    private LocalTime hourFrom;
    private LocalTime hourTo;
    private Integer daysReservationBefore;

    public static Attraction of(final AttractionEntity entity)
    {
        return Attraction.builder()
                .withAttractionId(entity.getAttractionId())
                .withTouristicPlaceId(entity.getTouristicPlaceId())
                .withAttractionType(AttractionType.fromValue(entity.getAttractionType()))
                .withName(entity.getName())
                .withDescription(entity.getDescription())
                .withPrice(entity.getPrice())
                .withPriceType(PriceType.fromValue(entity.getPriceType()))
                .withPrepayment(entity.getPrepayment())
                .withCancelReservationDays(entity.getCancelReservationDays())
                .withMaxPeopleNumber(entity.getMaxPeopleNumber())
                .withMaxItems(entity.getMaxItems())
                .withDateFrom(entity.getDateFrom())
                .withDateTo(entity.getDateTo())
                .withHourFrom(entity.getHourFrom())
                .withHourTo(entity.getHourTo())
                .withDaysReservationBefore(entity.getDaysReservationBefore())
                .build();
    }
}
